app/src/main/java/uk/ac/ebi/atlas/search/celltype/CellTypeSearchService.java

```diff
@@ -0,0 +1,29 @@
+package uk.ac.ebi.atlas.search.celltype;
+
+import com.google.common.collect.ImmutableSet;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.search.GeneSearchService;
+
+@Component
+@RequiredArgsConstructor
+public class CellTypeSearchService {
+
+    private final CellTypeSearchDao cellTypeSearchDao;
+    private final GeneSearchService geneSearchService;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CellTypeSearchService.class);
+
+    public ImmutableSet<String> search(ImmutableSet<String> geneIds, ImmutableSet<String> organismParts) {
+        if (geneIds.isEmpty()) {
```

Comment:
We can add not empty if condition first and then empty one please.
```
suggestion

if (!geneIds.isEmpty()) {
LOGGER.info("Searching organism parts for this gene ids: {}", geneIds.asList());
  return cellTypeSearchDao.searchCellTypes(geneSearchService.getCellIdsFromGeneIds(geneIds), organismParts);

}
else
{
   LOGGER.warn("Can't query for organism part as no gene IDs has given.");
            return ImmutableSet.of();
}
```

Reply:
@upendrakumbham These articles hopefully help you understand why fail-fast is a better way to write your code:
- https://www.martinfowler.com/ieeeSoftware/failFast.pdf
- https://en.wikipedia.org/wiki/Fail-fast
- https://softwareengineering.stackexchange.com/questions/254002/checking-negative-of-a-condition
Please let me know if it is not clear to you. We can discuss this F2F, if you prefer...
Thanks

Reply:
Agree with you. Thanks.